### PR TITLE
style(rustfmt): Run cargo fmt

### PR DESCRIPTION
**Pull request by klausi**
Every time I modify code to play with this library I have huge diffs because I use `cargo fmt` automatically. Can we cargo fmt here?

Note that I had to revert one change of cargo fmt regarding macros, will file an upstream bug report later.

### CODE REVIEW

**Comment by klausi** [comment]
Opened https://github.com/rust-lang-nursery/rustfmt/issues/2487

**Comment by Aaronepower** [comment]
@klausi Well I'd recommend against using `cargo fmt` when contributing. Especially as rustfmt isn't really close to stabilisation or really settled on what the default style should be. There are some corrections in here I agree with, I'm going to mark the changes I don't agree with and if you can patch those out of the PR I'd be happy to merge this in.

File: src/apps.rs [hunk 1]
```diff
@@ -54,15 +54,19 @@ pub enum Scopes {
...
-        write!(f, "{}", match *self {
-            All => "read%20write%20follow",
-            Follow => "follow",
-            Read => "read",
-            ReadFollow => "read%20follow",
-            ReadWrite => "read%20write",
-            Write => "write",
-            WriteFollow => "write%20follow"
-        })
+        write!(
```
Comment (Aaronepower) [inline src/apps.rs]: Please remove this one, it's makes it harder to read.

File: src/entities/attachment.rs [hunk 2]
```diff
@@ -20,15 +20,13 @@ pub struct Attachment {
...
+    #[serde(deserialize_with = "empty_as_none")]
     pub meta: Option<Meta>,
     /// Noop will be removed.
     pub description: Option<String>,
 }
 
-fn empty_as_none<'de, D: Deserializer<'de>>(val: D)
-    -> Result<Option<Meta>, D::Error>
-{
+fn empty_as_none<'de, D: Deserializer<'de>>(val: D) -> Result<Option<Meta>, D::Error> {
```
Comment (Aaronepower) [inline src/entities/attachment.rs]: Please remove this change, as a policy if a function signature is longer than 80 characters I place the return type, where clause and starting parens each on a new line.

File: src/lib.rs [hunk 3]
```diff
@@ -35,11 +35,13 @@
 #![cfg_attr(test, deny(warnings))]
 #![cfg_attr(test, deny(missing_docs))]
 
-#[macro_use] extern crate serde_derive;
-#[macro_use] extern crate serde_json as json;
 extern crate chrono;
 extern crate reqwest;
 extern crate serde;
+#[macro_use]
```
Comment (Aaronepower) [inline src/lib.rs]: Please remove this change, I prefer `#[macro_use]` on the same line as when it's sorted with vim's `:sort` function it will place all macro_use statements above the rest, whcich help prevents macros missing errors because they were defined after they were required.

File: src/lib.rs [hunk 4]
```diff
@@ -287,16 +289,14 @@ impl fmt::Display for Error {
 impl StdError for Error {
     fn description(&self) -> &str {
         match *self {
-            Error::Api(ref e) => {
-                e.error_description.as_ref().unwrap_or(&e.error)
-            },
+            Error::Api(ref e) => e.error_description.as_ref().unwrap_or(&e.error),
```
Comment (Aaronepower) [inline src/lib.rs]: Same as the function signature, anything beyond 80 characters should be truncated onto the next lines.

File: src/lib.rs [hunk 5]
```diff
@@ -314,38 +314,43 @@ pub struct ApiError {
...
+    fn from_registration<I>(
+        base: I,
+        client_id: I,
+        client_secret: I,
+        redirect: I,
+        token: I,
+        client: Client,
+    ) -> Self
+    where
+        I: Into<Cow<'static, str>>,
```
Comment (Aaronepower) [inline src/lib.rs]: If `-> Self` was on it's own line and the `where` clause was adjoined with the next line to make `where I: Into<Cow<'static, str>>,` I'd accept this.

File: src/lib.rs [hunk 6]
```diff
@@ -403,12 +408,10 @@ impl Mastodon {
         (delete) delete_status: "statuses/{}" => Empty,
     }
 
-    pub fn update_credentials(&self, changes: CredientialsBuilder)
-        -> Result<Account>
-    {
-
+    pub fn update_credentials(&self, changes: CredientialsBuilder) -> Result<Account> {
```
Comment (Aaronepower) [inline src/lib.rs]: Same as above, no function signatures longer than 80 characters on their own line.

File: src/lib.rs [hunk 7]
```diff
@@ -403,12 +408,10 @@ impl Mastodon {
...
         let url = self.route("/api/v1/accounts/update_credentials");
-        let response = self.client.patch(&url)
+        let response = self.client
+            .patch(&url)
```
Comment (Aaronepower) [inline src/lib.rs]: Revert this change please, semantically I prefer to start newline chaining functions with the first function with the object.

File: src/lib.rs [hunk 8]
```diff
@@ -426,8 +429,8 @@ impl Mastodon {
 
     /// Post a new status to the account.
     pub fn new_status(&self, status: StatusBuilder) -> Result<Status> {
-
-        let response = self.client.post(&self.route("/api/v1/statuses"))
+        let response = self.client
+            .post(&self.route("/api/v1/statuses"))
```
Comment (Aaronepower) [inline src/lib.rs]: Same as above.

File: src/lib.rs [hunk 9]
```diff
@@ -461,28 +464,26 @@ impl Mastodon {
...
-                    "&"
-                } else {
-                    "?"
-                };
+    pub fn statuses(
+        &self,
+        id: u64,
+        only_media: bool,
+        exclude_replies: bool,
+    ) -> Result<Vec<Status>> {
```
Comment (Aaronepower) [inline src/lib.rs]: Place the return type and opening parens on their own line.

File: src/lib.rs [hunk 10]
```diff
@@ -551,9 +552,7 @@ from! {
 
 // Convert the HTTP response body from JSON. Pass up deserialization errors
 // transparently.
-fn deserialise<T: for<'de> serde::Deserialize<'de>>(mut response: Response)
-    -> Result<T>
-{
+fn deserialise<T: for<'de> serde::Deserialize<'de>>(mut response: Response) -> Result<T> {
```
Comment (Aaronepower) [inline src/lib.rs]: Same as above.

**Comment by klausi** [comment]
Hm, then I think this pull request is not worth it because either we do full cargo fmt or not.

I found a workaround in my editor vscode by disabling cargo fmt on a per project basis.

Sorry for the noise!
